docs/source/metadata.rst

```diff
@@ -77,16 +81,66 @@ Accessing position metadata through the :class:`~libcst.MetadataDepedent`
 interface will return a :class:`~libcst.CodeRange` object. See
 :ref:`the above example<libcst-metadata-position-example>`.
 
-.. autoclass:: libcst.BasicPositionProvider
-.. autoclass:: libcst.SyntacticPositionProvider
+.. autoclass:: libcst.metadata.BasicPositionProvider
+.. autoclass:: libcst.metadata.SyntacticPositionProvider
 
 .. autoclass:: libcst.CodeRange
 .. autoclass:: libcst.CodePosition
 
 
 Expression Context Metadata
 ---------------------------
-.. autoclass:: libcst.ExpressionContextProvider
+.. autoclass:: libcst.metadata.ExpressionContextProvider
    :no-undoc-members:
 
-.. autoclass:: libcst.ExpressionContext
\ No newline at end of file
+.. autoclass:: libcst.metadata.ExpressionContext
+
+Scope Metadata
+--------------
+Scope is the block of naming binding. The bind name is not available
+after existing the bind block. Pyhon is
+`function-scoped <https://en.wikipedia.org/wiki/Scope_(computer_science)#Function_scope>`_.
+New scopes are created for classes and functions, but not other block constructs like
+conditional statements, loops, or try…except, don't create their own scope. E.g.::
+
+    ITERATIONS = 10
+    class Cls:
+        class_attribute = 20
+
+        def fn():
+            for i in range(ITERATIONS):
+                ...
+            return [
+                i for i in range(10)
+            ]
+    Cls().fn()
+
+The scopes of each name assignments and class/function definitions are visualized:
+
+.. image:: _static/img/python_scopes.svg
```

Comment:
- Use the png instead of the svg. The svg won't render properly if the user doesn't have the font we need. SVGs can use webfonts, which would solve this, but they're a pain to configure in inkscape, and a PNG is easier.

- Since this image contains the code above, I think including both the text and the graphic below it is redundant. We should just include the image.